contracts/crowdsale/price/CapStagedCrowdsale.sol

```diff
@@ -0,0 +1,93 @@
+pragma solidity ^ 0.4.24;
+
+import "../Crowdsale.sol";
+import "../../math/SafeMath.sol";
+import "../../ownership/Ownable.sol";
+import "../validation/CappedCrowdsale.sol";
+
+
+/**
+  * @title CapStagedCrowdsale
+  * @dev Extension of Crowdsale contract that changes the price of tokens regarding of raised ETH. Each stage of crowdsale is defined in StageLimit (cap limit for the stage) and stageRate (stage rate)
+  * Stages must be inserted from first to last with increasing stageLimit.
+  */
+contract CapStagedCrowdsale is CappedCrowdsale {
+  using SafeMath for uint256;
+
+  struct Stage {
+    uint256 limit;
+    uint256 rate;
+  }
+
+  Stage[] public stages;
+
+  /**
+   * @param _initialRate Number of tokens a buyer gets per wei - default parameter for crowdsale contract
```

Comment:
This is never used, so there should be a comment saying so. 

This might be a shortcoming of our crowdsales design. But let's see if we can give it a twist. What if we use the initialRate as the rate for the first stage. Every crowdsale must have at least one stage, and this way we could see normal crowdsales as having only one stage. This extension allows normal crowdsales to have more stages. If you add more stages, what you pass is the lower limit, the thing that triggers that stage instead of the one that ends it. This sounds more like a building block to me, as I was saying on my previous comment. But I'm just throwing an idea, feel free to say it's bad.

Reply:
Hi, 
i'm thinking how would this work as this contract is limited by wei raised for every stage and default crowdsale contract does not include any limits. User will still be forced to define first stage of 'CapStagedCrowdsale' with "rate" and "stageLimit" as every stage requires to have an upper limit. Maybe i can make this contract to work like this: Every stage is defined like now: stage limit and rate, but we would also have hardCap variable in this contract which is the upper limit of how much contract can raise. If for example: hardCap is set to 100000wei and last stage limit is 50000, then contract would take _initialRate when weiRaised is over last stage limit and still under hardCap?  Am i complicating too much?